### PR TITLE
Add phantomjs to web framework

### PR DESCRIPTION
**Pull request by LyzardKing**
This adds phantomjs downloaded from bitbucket (via https).
The medium and large tests work

### CODE REVIEW

File: tests/large/test_web.py [hunk 1]
```diff
@@ -115,6 +116,50 @@ def language_file_exists(self, language):
         return self.path_exists(os.path.join(self.installed_path, "dictionaries", "{}.aff".format(language)))
 
 
+class PhantomJSTests(LargeFrameworkTests):
+    """The default Go google compiler."""
```
Comment (didrocks) [inline tests/large/test_web.py]: It's not the Go compiler :)

File: tests/large/test_web.py [hunk 2]
```diff
@@ -115,6 +116,50 @@ def language_file_exists(self, language):
...
+        self.child.sendline("")
+        self.expect_and_no_warn("Installation done", timeout=self.TIMEOUT_INSTALL_PROGRESS)
+        self.wait_and_close()
+
+        self.assert_exec_exists()
+        self.assertTrue(self.is_in_path(self.exec_path))
+
+        # compile a small project
+        output = subprocess.check_output(self.command_as_list(compile_command)).decode()\
+            .replace('\r', '').replace('\n', '')
```
Comment (didrocks) [inline tests/large/test_web.py]: You can escape the \r\n by just slicing your output if you know they will be present. Or comparing with the assertEqual having \r\n.

File: tests/large/test_web.py (hunk 1 above)
Comment (LyzardKing) [inline tests/large/test_web.py]: I wonder how that got there...

File: tests/large/test_web.py (hunk 2 above)
Comment (LyzardKing) [inline tests/large/test_web.py]: It works if I instead use [:-1]
Should we then change it also in the go test (I have to admit I copied that...)

Comment (didrocks) [inline tests/large/test_web.py]: please, feel free to change wherever it makes sense :)
Just ensure this work both in a large tests and within the medium tests, as we don't have the exact same configuration there.

Comment (LyzardKing) [inline tests/large/test_web.py]: Sure ;) testing now for phantomjs in medium test

Comment (LyzardKing) [inline tests/large/test_web.py]: it works if I replace the mock command from the echo to:
`printf "hello, world\r"`
Is that ok?
For now I'll just fix it on phantom. then I'll do another pr to fix other tests later on.

Comment (didrocks) [inline tests/large/test_web.py]: Working for both medium and large tests? If so, yeah, feel free :)

Comment (LyzardKing) [inline tests/large/test_web.py]: And done. Should be good now.

**Comment by didrocks** [comment]
Merging then! Thanks a lot :)

**Pull request merged**